processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorsBuilder.java

```diff
@@ -65,22 +67,33 @@ public ProcessorsBuilder(String topic, TaskExtractor<T> taskExtractor, TaskExtra
 
     /**
      * Create new {@link ProcessorsBuilder} that consumes message from topic expecting tasks of type
-     * which can be parsed by valueParser.
+     * which can be parsed by task extractor.
      * @param topic the name of topic to consume.
      * @param taskExtractor the extractor to extract task of type {@link T} from message bytes.
      * @param <T> the type of instantiated tasks.
      * @return an instance of {@link ProcessorsBuilder}.
      */
     public static <T> ProcessorsBuilder<T> consuming(String topic, TaskExtractor<T> taskExtractor) {
         DefaultTaskExtractor<byte[]> innerExtractor = new DefaultTaskExtractor<>(bytes -> bytes);
-        TaskExtractor<T> retryTaskExtractor = bytes -> {
-            // Retry tasks are serialized as PB DecatonTaskRequest.
-            // First, deserialize PB from raw bytes.
-            DecatonTask<byte[]> retryTask = innerExtractor.extract(bytes);
-
-            // Original raw task bytes is stored in DecatonTaskRequest#serializedTask.
-            // Extract DecatonTask from DecatonTaskRequest#serializedTask using given taskExtractor.
-            DecatonTask<T> task = taskExtractor.extract(retryTask.taskData());
+        TaskExtractor<T> retryTaskExtractor = record -> {
+            // Retry tasks are serialized in different format depending on decaton client implementation.
+            // First, extract retry task from raw bytes.
+            DecatonTask<byte[]> retryTask = innerExtractor.extract(record);
+
+            // Restore ConsumerRecord which holds original task bytes as its value
+            ConsumerRecord<String, byte[]> restoredRecord =
```

Comment:
hm, now this implementation's getting bigger. shall we extract it out as a method or a class?